File: Serenity Practice/HosbitalSerenity.Web/Modules/Hosbital/Patients/PatientsColumns.cs
using HosbitalSerenity.Modules.Hosbital.Patients;

namespace HosbitalSerenity.Hosbital.Columns;

[ColumnsScript("Hosbital.Patients")]
[BasedOnRow(typeof(PatientsRow), CheckNames = true)]
public class PatientsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int PatientId { get; set; }
    [EditLink]
    public string PatientName { get; set; }
    public DateTime DateOfBirth { get; set; }
    public Gender Gender { get; set; }
    public string Address { get; set; }
    public int Age { get; set; }
    public string GovName { get; set; }
    public string CityName { get; set; }
    public int Cost { get; set; }
    public int LoyalityYears { get; set; }
    public int TotalCost { get; set; }
}

File: Serenity Practice/HosbitalSerenity.Web/Modules/ServerTypes/Hosbital/PatientsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { Gender } from "../Modules/Hosbital.Patients.Gender";
import { PatientsRow } from "./PatientsRow";

export interface PatientsColumns {
    PatientId: Column<PatientsRow>;
    PatientName: Column<PatientsRow>;
    DateOfBirth: Column<PatientsRow>;
    Gender: Column<PatientsRow>;
    Address: Column<PatientsRow>;
    Age: Column<PatientsRow>;
    GovName: Column<PatientsRow>;
    CityName: Column<PatientsRow>;
    Cost: Column<PatientsRow>;
    LoyalityYears: Column<PatientsRow>;
    TotalCost: Column<PatientsRow>;
}

export class PatientsColumns extends ColumnsBase<PatientsRow> {
    static readonly columnsKey = 'Hosbital.Patients';
    static readonly Fields = fieldsProxy<PatientsColumns>();
}

[Gender]; // referenced types